src/main/java/com/google/devtools/build/lib/rules/cpp/LinkCommandLine.java

```diff
@@ -296,39 +296,43 @@ CommandLine paramCmdLine() {
         doNotSplitLinkingCmdLine);
   }
 
-  public static void extractArgumentsForStaticLinkParamFile(
+  private static boolean isKnownAtArgument(String arg) {
```

Comment:
This feels like a huge hack, I would love to know if anyone else has ideas here. Theoretically we could force all users to pass `--linkopt=-Wl,-rpath,@executable_path` instead of `--linkopt=-rpath --linkopt=@executable_path`, but short of potentially breaking existing users, I don't see another path for this. In rules_swift we're able to check if the argument exists as a params file, but I don't think that would work in this context since the files wouldn't exist yet?

Reply:
cc @googlewalt I was told you might have some idea here

Reply:
I discussed with @trybka and others internally.  bazel is trying to get out of the business of understanding command lines, so we'd rather not to add more semantics there.

Suggested way to proceed:

* There is one instance of `-Xlinker -rpath -Xlinker @loader_path` in native code for AppleBinary -- I can fix that.
* Make sure that the existing external toolchain does not have any uses of prefix `@executable_path`, `@loader_path`, and `@rpath` arguments.  I don't believe it does?
* Officially, param files will not work with `-Xlinker -rpath -Xlinker @loader_path/@executable_path`.  The few (if any) affected people can either fix those flags, or turn off param files.

Then we can drop this patch.

Reply:
If you're happy with that I'm up for making those changes! I noticed those cases in bazel as well and they do seem trivially replacable. There are also some in the rules that we can fix https://github.com/bazelbuild/rules_apple/pull/941

Reply:
Updated, lmk what you think!